static/app/utils/orgRole.tsx

```diff
@@ -0,0 +1,12 @@
+import {OrgRole} from 'sentry/types';
+
+export function getTopOrgRole(memberOrgRoles: string[], orgRoleList: OrgRole[]) {
+  // sort by ascending index
+  memberOrgRoles.sort((a, b) =>
+    orgRoleList.findIndex(r => r.id === a) < orgRoleList.findIndex(r => r.id === b)
+      ? 1
+      : -1
+  );
+
+  return orgRoleList.find(r => r.id === memberOrgRoles[0]);
```

Comment:
The ternary operator seems redundant, you can simplify it by doing +- operations.

Rather than findIndex, it might also be more legible to transform the orgRoleList into a map.

```
const orgRoleMap = orgRoleList.reduce((acc, role, index) => {
  acc[r.id] = {index, role};
  return acc;
}, {});

memberOrgRoles.sort((a, b) => orgRoleMap[a.id].index - orgRoleMap[b.id].index);
return orgRoleMap[memberOrgRoles[0]].role;
```